api/python/cellxgene_census/src/cellxgene_census/experimental/ml/pytorch.py

```diff
@@ -77,82 +78,106 @@ class Stats:
     elapsed: int = 0
     """The total elapsed time in seconds for retrieving all batches"""
 
-    n_soma_batches: int = 0
-    """The number of batches retrieved"""
+    n_soma_chunks: int = 0
+    """The number of chunks retrieved"""
 
     def __str__(self) -> str:
-        return (
-            f"n_soma_batches={self.n_soma_batches}, n_obs={self.n_obs}, nnz={self.nnz}, "
-            f"elapsed={timedelta(seconds=self.elapsed)}"
-        )
+        return f"{self.n_soma_chunks=}, {self.n_obs=}, {self.nnz=}, " f"elapsed={timedelta(seconds=self.elapsed)}"
 
     def __add__(self, other: "Stats") -> "Stats":
         self.n_obs += other.n_obs
         self.nnz += other.nnz
         self.elapsed += other.elapsed
-        self.n_soma_batches += other.n_soma_batches
+        self.n_soma_chunks += other.n_soma_chunks
         return self
 
 
 @contextmanager
 def _open_experiment(
-    uri: str, aws_region: Optional[str] = None, soma_buffer_bytes: Optional[int] = None
+    uri: str,
+    aws_region: Optional[str] = None,
```

Comment:
side issue: this ties the implementation to AWS (or an AWS-like API). Is there a reason you pass only the `vfs.s3.region` value, rather than simply passing down the entire context config dictionary? In theory, the latter would allow portability to other environments, when/if we decide to host on non-AWS object stores.

Reply:
Well, it's a private method, and the region is determined from the `Experiment` passed in by the user (`experiment.context.tiledb_ctx.config().get("vfs.s3.region")`). So I think that sufficiently hides the AWS-specific details, right?

Reply:
My point was that it doesn't support non-AWS storage locations so it isn't very future-proof. Just transparently passing the entire config would be future-proof.

This is _totally_ a side issue, so feel free to ignore!